src/content/components/BugList/BugList.js

```diff
@@ -80,41 +74,64 @@ const priorityOptions = [
   },
 ];
 
-export class BugList extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      selectedBugs: {},
-      showResolved: props.showResolved,
-      editSelection: -1,
-      newIteration: currentIteration,
-      newPriority: "---",
+export const BugList = ({ ...props }) => {
+  const context = useContext(GlobalContext);
+  const [selectedBugs, setSelectedBugs] = useState({});
+  const [showResolved, setShowResolved] = useState(props.showResolved);
+  const [editSelection, setEditSelection] = useState(-1);
+  const [bugUpdating, setBugUpdating] = useState(false);
+
+  const mounted = useRef();
+  const bugsRef = useRef(props.bugs);
+  const iterationSelectRef = useRef(null);
+  const iterationRef = useRef(context.iterations.getIteration().number);
+  const priorityRef = useRef("---");
+
+  useEffect(() => {
+    if (!mounted.current) {
+      mounted.current = true;
+    } else if (
+      bugUpdating &&
+      JSON.stringify(props.bugs) != JSON.stringify(bugsRef.current)
+    ) {
+      setBugUpdating(false);
+      setEditSelection(-1);
+      setSelectedBugs({});
+    }
+
+    return () => {
+      bugsRef.current = props.bugs;
     };
-    this.onCheck = this.onCheck.bind(this);
-    this.onAllSelectedCheck = this.onAllSelectedCheck.bind(this);
-    this.onCheckShowResolved = this.onCheckShowResolved.bind(this);
-  }
+  }, [bugUpdating, props.bugs]);
+
+  const handleMenuOpen = useCallback(() => {
+    let { current } = iterationSelectRef;
+
+    requestAnimationFrame(() => {
+      current.focusedOptionRef?.scrollIntoView({ block: "nearest" });
+    });
+  }, []);
 
-  getRowClassName(bug) {
+  const getRowClassName = bug => {
     const classNames = [];
-    if (this.props.crossOutResolved && isBugResolvedOrMerged(bug)) {
+    if (props.crossOutResolved && isBugResolvedOrMerged(bug)) {
       classNames.push(styles.resolved);
     } else if (bug.assigned_to === "nobody@mozilla.org") {
       classNames.push(styles.unassigned);
     } else if (
-      this.props.bugzilla_email &&
-      bug.assigned_to === this.props.bugzilla_email
+      props.bugzilla_email &&
+      bug.assigned_to === props.bugzilla_email
     ) {
       classNames.push(styles.mine);
     }
     return classNames.join(" ");
-  }
+  };
 
-  renderColumn(columnId, bug) {
+  const renderColumn = (columnId, bug) => {
```

Comment:
these functions that return JSX should be their own component, if they're not part of a class component. the reason to use a class component would be that you don't want to waste resources re-defining this on every render. you could avoid that by making it a new component, but this will be a lot easier in general if you just leave it as a class component